src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java

```diff
@@ -84,6 +91,16 @@ public QueryResponseBean(QueryScreen queryScreen) {
                     );
             count++;
         }
+
+        OrderedHashtable<String, QueryGroup> queryGroupMap = queryScreen.getGroupHeaders();
+        groupHeader = new OrderedHashtable<>();
+        for (Map.Entry<String, QueryGroup> entry : queryGroupMap.entrySet()) {
+            String key = entry.getKey();
+            QueryGroup queryGroupItem = entry.getValue();
+            String text = queryGroupItem.getDisplay().getText().evaluate(ec);
+            groupHeader.put(key, text);
+        }
```

Comment:
[11d6ed6](https://github.com/dimagi/formplayer/pull/1511/commits/11d6ed6732f1c192a4c2d7ac19a66aba435f2f64) and [15a0d9e](https://github.com/dimagi/commcare-core/pull/1389/commits/15a0d9ecbc5783cc28d53531e81312208866f4f9)

Can you explain what the responsibility of `queryScreen` is? Why would the preference be to put the logic in there instead of extracting it to another function in `queryResponseBean`? I would imagine `queryResponesBean` is responsible for generating the data in the response.

Reply:
`queryResponseBean` is supposed to me more of a response model wrapper class over `screen` classes with `screen` classes being the classes that contain any evaluation logic. We don't follow it strictly but that's the idea atleast in theory.